papers/pydra/paper.rst

```diff
@@ -0,0 +1,467 @@
+
+This is schematically represented in Fig. :ref:`ndspl3`
+
+.. figure:: nd_spl_3-crop.pdf
+   :figclass: h!
+   :scale: 75%
+
+   Diagram representing a Task with two input fields and an outer splitter. :label:`ndspl3`
+
+
+In addition to the splitting the input, *Pydra* supports grouping or combining the output together.
+Taking as an example the simple *Task* represented in Fig. :ref:`ndspl1`, in some application
+it could be useful to combine all the outputs at the end.
+In order to do it *Task* has to have so called *combiner* that could be set by calling ``combine`` method.
+This could be written as follow:
+
+
+.. math::
+   :type: eqnarray
+
+   \textcolor{red}{\mathnormal{S} = x} &:& x=[x_1, x_2, ..., x_n] \longmapsto x=x_1, x=x_2, ..., x=x_n \\
+   \textcolor{red}{\mathnormal{C} = x} &:& out(x_1), out(x_2), ...out(x_n) \longmapsto out=[out(x_1), out(x_2), ...out(x_n)]
+
+
+Again, in the situation where input has multiple fields, there are various way of combining the output.
+Taking as an example *Task* represented in Fig. :ref:`ndspl4`, it could be useful to combine all the outputs
+for one specific values of *x* and all the values of *y*.
+The combining operation could be written as follow:
+
+.. math::
+   :type: eqnarray
+
+   \textcolor{red}{\mathnormal{C} = y} &:& out(x_1, y1), out(x_1, y2), ...out(x_n, y_m) \\
+    &\longmapsto& [[out(x_1, y_1), ..., out(x_1, y_m)], \\
+    && ..., \\
+    && [out(x_n, y_1), ..., out(x_n, y_m)]]
+
+And is represented in Fig. :ref:`ndspl3comb1` (todo: should probably change a,b to x,y)
+
+
+.. figure:: nd_spl_3_comb1-crop.pdf
+   :figclass: h!
+   :scale: 75%
+
+   Diagram representing a Task with two input fields, an outer splitter and a combiner. :label:`ndspl3comb1`
+
+It is also possible to combine all values of *x* for a specifc values of *y*, but it is also possible
+to combine all the values by providing a list of fields to the combiner.
+When a full combiner is set, i.e. all fields are in the combiner, the output is a one dimensional list:
+
+.. math::
+   :type: eqnarray
+
+   \textcolor{red}{\mathnormal{C} = [x, y]} : out(x_1, y1), ...out(x_n, y_m)
+    \longmapsto [out(x_1, y_1), ..., out(x_n, y_m)]
+
+
+And is represented in Fig. :ref:`ndspl3comb3` (todo: should probably change a,b to x,y)
+
+
+.. figure:: nd_spl_3_comb3-crop.pdf
+   :figclass: h!
+   :scale: 75%
+
+   Diagram representing a Task with two input fields, an outer splitter and a full combiner. :label:`ndspl3comb3`
+
+
+These are the basic examples of *splitters* and *combiners*, but *Pydra* allows for mixing
+*splitters* and *combiners* on various level.
+They could be set on a single *Task* level, or on *Workflow* level.
+They could be also passed from one *Task* to the followings within a *Workflow*.
+
+
+Global Cache
+============
+
+
+
+Applications and Examples
+-------------------------
+
+Machine Learning: Model Comparison
+==================================
+
+
+.. code-block:: python
+
+  ml example TODO
```

Comment:
Just use the ml example from the [pycon poster](https://docs.google.com/presentation/d/10tS2I34rS0G9qz6v29qVd77OUydjP_FdBklrgAGmYSw/edit?usp=sharing)?    I'll add the code below and include an example workflow that uses a public dataset